src/entities/fractions/priced-token-amount.ts

```diff
@@ -0,0 +1,17 @@
+import { BigintIsh } from '../../constants'
+import { PricedToken } from '../priced-token'
+import { CurrencyAmount } from './currencyAmount'
+
+export class PricedTokenAmount extends CurrencyAmount {
+  public readonly token: PricedToken
+
+  // amount _must_ be raw, i.e. in the native representation
+  public constructor(token: PricedToken, amount: BigintIsh) {
+    super(token, amount)
```

Comment:
Is there a way to check if `amount` is actually in native representation?

Reply:
Sorry, missed it. The amount is a simple integer number, so it has no info about the representation. We could check it by looking at the `PricedToken`. But the goal here is to have generic stuff, where the quote currency of the price is generic.